### PR TITLE
Limiting Object Depth Traversal

### LINKED ISSUE #7: Create the ability to limit the depth of object traversal

### PR DESCRIPTION
**Pull request by brettminnie**
Will close #7

### CODE REVIEW

File: src/BreakfastSerializer/Serializer.php [hunk 1]
```diff
@@ -63,6 +70,55 @@ public function getDepth()
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isWithinBounds()
+    {
+        $isValid = (self::MAX_DEPTH_NOT_SET === $this->maxDepth)
+            ? true : ($this->currentDepth <= $this->maxDepth)
+                ? true : false;
```
Comment (eddiejaoude) [inline src/BreakfastSerializer/Serializer.php]: nested ternary?

Comment (brettminnie) [inline src/BreakfastSerializer/Serializer.php]: Why yes it is

A thing of beauty is it not :)

On Fri, Apr 24, 2015 at 1:28 PM, Eddie Abou-Jaoude <notifications@github.com

Comment (eddiejaoude) [inline src/BreakfastSerializer/Serializer.php]: Haha...I am not even justifying that with a reply! :wink:

Comment (brettminnie) [inline src/BreakfastSerializer/Serializer.php]: real programmers write everything in nested ternaries

```c++
int i = 5;
string result = i % 2 == 0 ? "a" : i % 3 == 0 ? "b" : i % 5 == 0 ? "c" : i
% 7 == 0 ? "d" : "e";
```

On Fri, Apr 24, 2015 at 1:34 PM, Eddie Abou-Jaoude <notifications@github.com

**Comment by brettminnie** [comment]
@eddiejaoude when you get a moment can you cast a beady eye on this and if happy press the go button

**Comment by eddiejaoude** [comment]
Just 1 comment, not a show stopper, happy to merge?

**Comment by brettminnie** [comment]
Cool, I'll get on with the next pass and then we can test it on some data

**Comment by brettminnie** [comment]
:+1:


On Sun, Apr 26, 2015 at 9:34 AM, Eddie Abou-Jaoude <notifications@github.com

**Pull request merged**